armada/api/tiller_controller.py

```diff
@@ -13,53 +13,74 @@
 
-from armada.handlers.tiller import Tiller as tillerHandler
+from armada import api
+from armada.handlers.tiller import Tiller
 
-LOG = logging.getLogger(__name__)
-CONF = cfg.CONF
-DOMAIN = "armada"
-
-logging.setup(CONF, DOMAIN)
-
-class Status(object):
+class Status(api.BaseResource):
     def on_get(self, req, resp):
         '''
         get tiller status
         '''
-        message = "Tiller Server is {}"
-        if tillerHandler().tiller_status():
-            resp.data = json.dumps({'message': message.format('Active')})
-            LOG.info('Tiller Server is Active.')
-        else:
-            resp.data = json.dumps({'message': message.format('Not Present')})
-            LOG.info('Tiller Server is Not Present.')
+        try:
+            policy_action = 'chart_installer:query_status'
+            ctx = req.context
+
+            if not self.check_policy(policy_action, ctx):
+                self.access_denied(req, resp, policy_action)
+                return
 
-        resp.content_type = 'application/json'
-        resp.status = HTTP_200
+            message = {
+                'tiller': Tiller().tiller_status()
+            }
 
-class Release(object):
+            if message.get('tiller'):
+                resp.data = json.dumps(message)
+                resp.status = HTTP_200
+            else:
+                resp.data = json.dumps(message)
+                resp.status = HTTP_404
+
+            resp.content_type = 'application/json'
+        except Exception:
+            self.error(req.contex, "Unable to find resources")
+            self.return_error(
+                resp,
+                HTTP_404,
+                message="Unable to find resources"
+            )
+
+
+class Release(api.BaseResource):
     def on_get(self, req, resp):
         '''
         get tiller releases
         '''
-        # Get tiller releases
-        handler = tillerHandler()
 
-        releases = {}
-        for release in handler.list_releases():
-            releases[release.name] = release.namespace
+        try:
+
+            policy_action = 'chart_installer:read_releases'
+            ctx = req.context
+
+            if not self.check_policy(policy_action, ctx):
+                self.access_denied(req, resp, policy_action)
+                return
+
+            # Get tiller releases
+            handler = Tiller()
+            releases = {}
+            for release in handler.list_releases():
+                releases[release.name] = release.namespace
 
-        resp.data = json.dumps({'releases': releases})
-        resp.content_type = 'application/json'
-        resp.status = HTTP_200
+            resp.data = json.dumps({'releases': releases})
+            resp.content_type = 'application/json'
+            resp.status = HTTP_200
+        except Exception:
+            self.error(req.contex, "Unable to find Armada Releases")
+            self.return_error(
+                resp,
+                HTTP_404,
```

Comment:
Falcon generally handles 404 codes on their own when an endpoint does not exist, and this could possibly be a confusing status code to receive if no releases are available. 

If it is not an error that no releases were found, then a 2xx seems fitting because the request was successful and the results are not unexpected. If this is an unexpected error, it is a server error rather than a client error, and 5xx makes more sense.